lib/ecmascript.ts

```diff
@@ -372,9 +370,9 @@ export const ES = ObjectAssign({}, ES2020, {
     const weeks = ES.ToInteger(match[4]) * sign;
     const days = ES.ToInteger(match[5]) * sign;
     const hours = ES.ToInteger(match[6]) * sign;
-    let fHours = match[7];
+    let fHours: number | string = match[7];
```

Comment:
It isn't a number, but it is typed as such to "cheat" and allow that variable to be reassigned to a number below (presumably to save on memory allocation?).

Naively, I would have pulled these out in `const`s and then create a new variable to store the numeric conversion of this value, but I was attempting to avoid actual code changes (vs just type-space additions) so the runtime behaviour wouldn't change very much.

Reply:
Sounds good, thanks.